gdnative-core/src/nativescript/class.rs

```diff
@@ -321,6 +321,8 @@ impl<T: NativeClass> Instance<T, Unique> {
                 std::ptr::null_mut(),
             );
 
+            debug_assert!(class_registry::is_class_registered::<T>(), "`{type_name}` must be registered before it can be used; call `handle.add_class::<{type_name}>()` in your `nativescript_init` callback", type_name = std::any::type_name::<T>());
```

Comment:
Now that this message is here, perhaps the message on L335 can be changed to reflect this? Maybe `the engine should return a base object of the correct type`?

Reply:
Alrighty, I can make that change.

Reply:
Resolved with aeed339